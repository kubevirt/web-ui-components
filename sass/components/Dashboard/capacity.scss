.kubevirt-capacity__items {
  display: flex;
  justify-content: space-evenly;
  flex-wrap: wrap;
}

.kubevirt-capacity__item {
  display: flex;
  flex-direction: column;
  text-align: center;
  justify-content: space-between;
  flex: 0 1 120px;
}

.kubevirt-capacity__item-title {
  margin-top: 0;
}

.kubevirt-capacity__item-description {
  font-size: 0.6875em;
}

.kubevirt-capacity__item .donut-title-big-pf {
  font-size: 24px;
}

/* stylelint-disable plugin/selector-bem-pattern */
.kubevirt-capacity__graph {
  display: flex;
  flex-direction: column;

  svg {
    overflow: visible;
  }
}

.kubevirt-capacity__card {
  height: 100%;
}
/* stylelint-enable */
