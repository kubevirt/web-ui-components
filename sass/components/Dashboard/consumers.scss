.kubevirt-consumers__bar {
  height: 1.2em;
  margin-bottom: 0.6em;
  text-align: left;
}

.kubevirt-consumers__bar-progress {
  height: 0.6em;
  margin: 0.4em 0;
}

.kubevirt-consumers__bar-column-progress {
  padding: 0;
}

.kubevirt-consumers__bar-description {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  font-size: 0.8em;
}

.kubevirt-consumers__bar-label {
  text-align: right;
  font-size: 0.8em;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.kubevirt-consumers__metric-description {
  text-align: left;
  padding: 0;
}

.kubevirt-consumers__metric-type {
  text-align: left;
}

.kubevirt-consumers__filters {
  display: flex;
  padding-bottom: 0.6em;
  border-bottom: 1px solid #d1d1d1;
  justify-content: space-between;
}

.kubevirt-consumers__results {
  text-align: center;
}

.kubevirt-consumers__description {
  margin-top: 0.6em;
  margin-bottom: 0.6em;
}
