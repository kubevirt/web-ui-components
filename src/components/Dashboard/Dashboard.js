import React from 'react';
import PropTypes from 'prop-types';

const Dashboard = ({ children }) => <div>{children}</div>;

Dashboard.defaultProps = {
  children: null,
};

Dashboard.propTypes = {
  children: PropTypes.node,
};

export default Dashboard;
