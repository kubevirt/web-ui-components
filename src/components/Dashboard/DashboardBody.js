import React from 'react';
import PropTypes from 'prop-types';

import classNames from 'classnames';

const DashboardBody = ({ className, children }) => (
  <div className={classNames('kubevirt-dashboard__body', className)}>{children}</div>
);

DashboardBody.defaultProps = {
  className: null,
  children: null,
};

DashboardBody.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node,
};

export default DashboardBody;
