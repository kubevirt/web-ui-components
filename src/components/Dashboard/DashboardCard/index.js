export { default as DashboardCard } from './DashboardCard';
export { default as DashboardCardBody } from './DashboardCardBody';
export { default as DashboardCardHeader } from './DashboardCardHeader';
export { default as DashboardCardTitle } from './DashboardCardTitle';
export { default as DashboardCardTitleHelp } from './DashboardCardTitleHelp';
export { DashboardCardTitleSeeAll } from './DashboardCardTitleSeeAll';
