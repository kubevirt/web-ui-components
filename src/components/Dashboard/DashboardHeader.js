import React from 'react';
import PropTypes from 'prop-types';

const DashboardHeader = ({ children }) => (
  <div>
    <h1>{children}</h1>
  </div>
);

DashboardHeader.defaultProps = {
  children: null,
};

DashboardHeader.propTypes = {
  children: PropTypes.node,
};

export default DashboardHeader;
