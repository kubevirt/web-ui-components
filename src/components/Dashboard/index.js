export { default as Dashboard } from './Dashboard';
export { default as DashboardHeader } from './DashboardHeader';
export { default as DashboardBody } from './DashboardBody';
export * from './DashboardCard';
