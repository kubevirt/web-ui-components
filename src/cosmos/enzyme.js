import createContext from 'react-cosmos-test/enzyme';

import paths from '../../config/paths';

export default function createTestContext(args) {
  args.proxies = args.proxies || []; // workaround of react-cosmos-test bug
  return createContext({
    ...args,
    cosmosConfigPath: `${paths.config}/cosmos.config.js`,
  });
}
