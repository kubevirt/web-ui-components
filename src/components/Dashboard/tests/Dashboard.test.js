import React from 'react';
import { shallow } from 'enzyme';

import Dashboard from '../Dashboard';

const testDashboard = () => <Dashboard>content</Dashboard>;

describe('<Dashboard />', () => {
  it('renders correctly', () => {
    const component = shallow(testDashboard());
    expect(component).toMatchSnapshot();
  });
});
