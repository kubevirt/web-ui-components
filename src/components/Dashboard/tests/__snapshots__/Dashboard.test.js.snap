// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> renders correctly 1`] = `
<div>
  content
</div>
`;
